integration_tests/src/test/kotlin/integration/CustomShardingIT.kt

```diff
@@ -0,0 +1,186 @@
+import utils.asOutputReport
+import utils.assertContainsUploads
+import utils.assertCostMatches
+import utils.assertExitCode
+import utils.assertTestFail
+import utils.assertTestPass
+import utils.assertTestResultContainsWebLinks
+import utils.findTestDirectoryFromOutput
+import utils.json
+import utils.loadAsTestSuite
+import utils.multipleFailedTests
+import utils.multipleSuccessfulTests
+import utils.removeUnicode
+import utils.toJUnitXmlFile
+import utils.toOutputReportFile
+import java.nio.file.Paths
+
+class CustomShardingIT {
+    private val name = this::class.java.simpleName
+
+    private val jsonMapper by lazy { JsonMapper().registerModule(KotlinModule()) }
+
+    @get:Rule
+    val root = TemporaryFolder()
+
+    @Test
+    fun `flank custom sharding -- android`() {
+
+        val templateConfigPath =
+            "$CONFIGS_PATH/flank_android_custom_sharding.yml"
+
+        val customShardingPath = root.newFile("custom_sharding.json").also {
+            it.writeText(jsonMapper.writeValueAsString(customSharding))
+        }.absolutePath
+
+        val config = root.newFile("flank.yml").also {
+            it.writeText(
+                Paths.get(templateConfigPath)
+                    .toFile()
+                    .readText()
+                    .replace("{{PLACEHOLDER}}", customShardingPath)
+            )
+        }.absolutePath
+
+        val result = FlankCommand(
+            flankPath = FLANK_JAR_PATH,
+            ymlPath = config,
+            params = androidRunCommands
+        ).run("./", name)
+
+        assertExitCode(result, 10)
+
+        val resOutput = result.output.removeUnicode()
+
+        assertContainsUploads(
+            resOutput,
+            "app-multiple-success-debug-androidTest.apk",
+            "app-multiple-error-debug-androidTest.apk",
+            "MainActivity_robo_script.json"
+        )
+
+        resOutput.findTestDirectoryFromOutput().toJUnitXmlFile().loadAsTestSuite().run {
+            assertTestResultContainsWebLinks()
+            assertTestPass(multipleSuccessfulTests)
+            assertTestFail(multipleFailedTests)
+
+            assertThat(testSuites.size).isEqualTo(9)
+            assertThat(testSuites.filter { it.name == "junit-ignored" }.size).isEqualTo(1)
+        }
+
+        val outputReport = resOutput.findTestDirectoryFromOutput().toOutputReportFile().json().asOutputReport()
+
+        assertThat(outputReport.error).isEmpty()
+        assertThat(outputReport.cost).isNotNull()
+
+        outputReport.assertCostMatches()
+
+        assertThat(outputReport.testResults.count()).isEqualTo(4)
+        assertThat(outputReport.weblinks.count()).isEqualTo(4)
+
+        val testsResults = outputReport.testResults
+            .map { it.value }
+            .map { it.testAxises }
+            .flatten()
+
+        assertThat(testsResults.sumBy { it.testSuiteOverview.failures }).isEqualTo(5)
+        assertThat(testsResults.sumBy { it.testSuiteOverview.total }).isEqualTo(41)
```

Comment:
there is a helper function for it please see `assertTestCountMatches` in file `IntegrationTestUtils`

Reply:
Hm..I think I can't use it here. It's a similar case to the one in `MultipleApksIT`

Reply:
ohh..sorry